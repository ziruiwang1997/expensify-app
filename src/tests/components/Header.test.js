import React from 'react';
import { shallow } from 'enzyme';
//import ReactShallowRenderer from 'react-test-renderer/shallow';
import Header from '../../components/Header';
import toJson from 'enzyme-to-json';
test('shoul', () => {
  const wrapper = shallow(<Header/>);
  expect(toJson(wrapper)).toMatchSnapshot();

})