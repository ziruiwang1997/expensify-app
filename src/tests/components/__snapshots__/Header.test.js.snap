// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    startLogout={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Expensify
</h1>,
        <NavLink
          activeClassName="is-active"
          exact={true}
          to="/"
>
          Dashboard
</NavLink>,
        <NavLink
          activeClassName="is-active"
          to="/create"
>
          Create Expense
</NavLink>,
        <button
          onClick={[Function]}
>
          Logout
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Expensify",
        },
        "ref": null,
        "rendered": "Expensify",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "is-active",
          "children": "Dashboard",
          "exact": true,
          "to": "/",
        },
        "ref": null,
        "rendered": "Dashboard",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "is-active",
          "children": "Create Expense",
          "to": "/create",
        },
        "ref": null,
        "rendered": "Create Expense",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Logout",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Logout",
        "type": "button",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Expensify
</h1>,
          <NavLink
            activeClassName="is-active"
            exact={true}
            to="/"
>
            Dashboard
</NavLink>,
          <NavLink
            activeClassName="is-active"
            to="/create"
>
            Create Expense
</NavLink>,
          <button
            onClick={[Function]}
>
            Logout
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Expensify",
          },
          "ref": null,
          "rendered": "Expensify",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "is-active",
            "children": "Dashboard",
            "exact": true,
            "to": "/",
          },
          "ref": null,
          "rendered": "Dashboard",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "is-active",
            "children": "Create Expense",
            "to": "/create",
          },
          "ref": null,
          "rendered": "Create Expense",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Logout",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Logout",
          "type": "button",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
          "getChildContext": Object {
            "calledByRenderer": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
